lib/Public/ContentManager.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Nextcloud - ContextChat
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Anupam Kumar <kyteinsky@gmail.com>
+ * @copyright Anupam Kumar 2024
+ */
+
+namespace OCA\ContextChat\Public;
+
+use OCA\ContextChat\BackgroundJobs\InitialContentImportJob;
+use OCA\ContextChat\BackgroundJobs\SubmitContentJob;
+use OCA\ContextChat\Db\QueueContentItem;
+use OCA\ContextChat\Db\QueueContentItemMapper;
+use OCA\ContextChat\Service\LangRopeService;
+use OCA\ContextChat\Service\ProviderConfigService;
+use OCP\BackgroundJob\IJobList;
+use OCP\Server;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\NotFoundExceptionInterface;
+use Psr\Log\LoggerInterface;
+
+class ContentManager {
+	public function __construct(
+		private IJobList $jobList,
+		private ProviderConfigService $configService,
+		private LangRopeService $service,
+		private QueueContentItemMapper $mapper,
+		private LoggerInterface $logger,
+	) {
+	}
+
+	/**
+	 * @param class-string<IContentProvider> $providerClass
+	 * @return void
+	 * @since 1.1.0
+	 */
+	public function registerContentProvider(string $providerClass): void {
+		try {
+			/** @var IContentProvider */
+			$providerObj = Server::get($providerClass);
+		} catch (ContainerExceptionInterface | NotFoundExceptionInterface $e) {
+			$this->logger->warning('Could not register content provider', ['exception' => $e]);
+			return;
+		}
+
+		if ($this->configService->hasProvider($providerObj->getAppId(), $providerObj->getId())) {
+			return;
+		}
+
+		$this->configService->updateProvider($providerObj->getAppId(), $providerObj->getId(), $providerClass);
+
+		if (!$this->jobList->has(InitialContentImportJob::class, $providerClass)) {
```

Comment:
This should also check whether the configService has an initialized flag for this provider. Otherwise, we start spamming background jobs :)

Reply:
it is checked inside the import job. And we don't duplicate the jobs by matching the provider class string.

Reply:
> And we don't duplicate the jobs by matching the provider class string

Ah, right. This would still run everytime the register function is called, though, which is every request, so we would always have exactly one initial import job for each provider at all times, right?

Reply:
hmm, yeah I can see that happening. Fix coming right up!

Reply:
but wait, the provider would then already be in the providers config list and we won't get further with the process due to this check:
```php
if ($this->configService->hasProvider($providerObj->getAppId(), $providerObj->getId())) {
    return;
}
```

Reply:
Ahh, right. Sorry, I didn't see that! You're right :)